bin/desi_qso_qn_afterburner

```diff
@@ -162,39 +158,61 @@ def collect_redshift_with_new_RR_run(spectra_name, targetid, z_prior, param_RR):
 
             redshift = rr['Z'][correct_index]
             err_redshift = rr['ZERR'][correct_index]
+            chi2 = rr['CHI2'][correct_index]
             coeffs = np.zeros((redshift.size, 10))
             coeffs[:, :4] = rr['COEFF'][correct_index]
 
-        return redshift, err_redshift, coeffs
-
-    write_prior_for_RR(targetid, z_prior, filename_priors)
-
-    # Info: in the case where targetid is -7008279, we cannot use it at first element of the targetid list
-    # otherwise RR required at least one argument for --targetids .. (it is a known problem in python this comes from -)
-    # see for example https://webdevdesigner.com/q/can-t-get-argparse-to-read-quoted-string-with-dashes-in-it-47556/
-    # To figure out with this, we just need to add a space before the -
-
-    argument_for_rerun_RR = ['--infiles', spectra_name,
-                             '--templates', template_filename,
-                             '--targetids', ' ' + ",".join(reversed(targetid.astype(str))),
-                             '--priors', filename_priors,
-                             '--details', filename_output_rerun_RR,
-                             '--outfile', filename_redrock_rerun_RR]
-    # NEW RUN RR
-    log.info('Running redrock with priors on selected targets')
-    print()
-    rrdesi(argument_for_rerun_RR)
-    print()
-    log.info('Done running redrock')
-
-    # Extract information from the new run of RR
-    redshift, err_redshift, coeffs = extract_redshift_info_from_RR(filename_redrock_rerun_RR, targetid)
-
-    if param_RR['delete_RR_output'] == 'True':
-        log.debug("Remove output from the new run of RR")
-        os.remove(filename_priors)
-        os.remove(filename_output_rerun_RR)
-        os.remove(filename_redrock_rerun_RR)
+        return redshift, err_redshift, chi2, coeffs
+
+    # define the output
+    redshift, err_redshift, chi2, coeffs = np.zeros(targetid.size), np.zeros(targetid.size), np.inf * np.ones(targetid.size), np.zeros((targetid.size, 10))
+
+    # make an independant run for each quasar templates to circumvent some problem from redrock:
+    # 1] cannot give two templates in redrock as input, only one or a directory
+    # 2] problem with prior and template redshift range .. --> give zero result and redrock stop
+    for template_filename in param_RR['templates_filename']:
+        filename_priors = param_RR['filename_priors']
+        filename_output_rerun_RR = param_RR['filename_output_rerun_RR']
+        filename_redrock_rerun_RR = param_RR['filename_redrock_rerun_RR']
+
+        # find redshift range of the template
+        redshift_template = fitsio.FITS(template_filename)['REDSHIFTS'][:]
+        zmin, zmax = np.min(redshift_template), np.max(redshift_template)
+        sel = (z_prior >= zmin) & (z_prior <= zmax)
+
```

Comment:
What happens if no targets pass the selection?  I was expecting something like
```python
if np.all(sel == False):
    contintue
```
instead of proceeding with a redrock fit of no targets, which may work but might also produce messy log error messages.

Reply:
Good  point .. I just experiment this, I had a failure when I pass the specific targetid in redrock. I add the selection with `sel.sum() != 0` (do it before reading your comment.